conformance/tests/listenerset-protocol-conflict.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayxv1a1 "sigs.k8s.io/gateway-api/apisx/v1alpha1"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/pkg/features"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, ListenerSetProtocolConflict)
+}
+
+var ListenerSetProtocolConflict = suite.ConformanceTest{
+	ShortName:   "ListenerSetProtocolConflict",
+	Description: "Listener Set listener with protocol conflict with a Gateway listener",
```

Comment:
This one also falls under the precedence rules, so the requests to the conflicted listener should succeed, and MUST be http requests.

The conflicted Listener on the ListenerSet should be not Accepted, and the Listener on the Gateway should be marked as "Conflicted".

Maybe you could add a HTTPRoute that attaches to the conflicted Listener, and validate that traffic via that HTTPRoute should _not_ flow (maybe a different path?), and another that goes on the non-conflicted ListenerSet listener, and validates that traffic via that HTTPRoute _should_ flow.